### PR TITLE
Added grading infra

### PR DESCRIPTION
**Pull request by Njanderson**
Initial version of grading infrastructure used to grade the Spring 2018 offering of CSE 452 using these labs.

### CODE REVIEW

**Comment by Njanderson** [comment]
Everything that you said should be addressed. I also added a TODO.md file because I know this grader isn't perfect. Anyway, now the results of the grading should be sitting in a merged.json and merged directory, instead of making people do any merging themselves.

File: grading/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# 452 Grading
+
+## Files
+- config.json: contains the test configuration (what tests to run, where they are, where the student submissions are, what servers to use, where the jars/etc are). There are some old configuration parameters for git integration, but I didn't finish that up.
+- distributor.py: splits up the students into groups for each of the hosts that the tests will be run on. Rsyncs the students to grade onto each host and starts to run grader.py.
+- grader.py: grades on a local machine with many command line parameters corresponding to all of the required items.
+- test-resources: These are the test resources needed to compile: things like the framework and test jars, also the Makefile that run-tests.py uses.
+- Grades: Where I will move the grades after I invoke the grader.
+- Grade Exporter: A script that can import the grades into catalyst. Look at the [readme](grade_exporter/README.md) for more info.
```
Comment (emichael) [inline grading/README.md]: This no longer exists, right?

Comment (Njanderson) [reply grading/README.md]: Yeah, let me take that out and perhaps add to the README a bit more.

**Comment by emichael** [comment]
Looking good! Yeah, as soon as the broken links are fixed, I'll merge it in. I bet others will find these scripts useful. Thanks again!

**Comment by Njanderson** [comment]
Last thing - let me know if you want me to remove my contact information.

**Comment by emichael** [comment]
Nope, that's totally fine!

Just do one quick `grep -R 452`. After that I think we're good.

**Comment by Njanderson** [comment]
Okay, I removed the 2 452's that were found. I also added another script description for another useful script.

**Pull request merged**